mobile/src/screens/OnboardAppRouter/index.js

```diff
@@ -1,72 +1,95 @@
-import React from 'react'
-import {
-  Text,
-  View,
-  Image,
-  SafeAreaView
-} from 'react-native'
-import { ScaledSheet } from 'react-native-size-matters'
-import { colors, fonts, fontFamilies } from '../../utils/theme'
-import RoundButton from '../../components/RoundButton'
-import PurpleBackground from '../../components/PurpleBackground'
+import React, { useEffect } from "react"
+import { Text, View, Image, SafeAreaView } from "react-native"
+import { ScaledSheet } from "react-native-size-matters"
+import { colors, fonts, fontFamilies } from "../../utils/theme"
+import RoundButton from "../../components/RoundButton"
+import PurpleBackground from "../../components/PurpleBackground"
 // import { Auth } from 'aws-amplify'
 
-export default function OnboardAppRouter({ navigation }) {
-  const [user, setUser] = React.useState(null)
+export default function OnboardAppRouter({
+    navigation,
+    gameSession,
+    teamId,
+    teamMember,
+}) {
+    const [user, setUser] = React.useState(null)
 
-  function handleJoinGame() {
-    // screenProps.handleSetAppState('deviceSettings', { role: 'teacher' });
-    // if (user == null) {
-    //   setTimeout(() => navigation.navigate('StudentFirst', { user: false }), 250)
-    // } else {
-    //   setTimeout(() => navigation.navigate('StudentFirst', { user: true }), 250)
-    // }
-    navigation.navigate('StudentFirst')
-  }
+    useEffect(() => {
+        if (gameSession?.currentState === "TEAMS_JOINING") {
```

Comment:
We have `GameSessionState`, we should use that instead of hardcoding strings again here